### ISSUE TITLE
Button in loading state only prevents pointer events and is still interactive

### ISSUE THREAD

**Issue by lscheibel** [mantinedev/mantine#4176]
### What package has an issue

@mantine/core

### Describe the bug

When adding the loading prop to the button component, the button is not clickable anymore. This is achieved using the css property `pointer-events: none`. The button itself however is not disabled and still "clickable" by using for example `tab + enter` or `$button.click()`.

**Expected:** Since the button suggests that it is not interactive anymore, I'd expect the button to act as if it were disabled.
**Actual:** Using keyboard navigation or other less common means, I can still interact with the button.

### What version of @mantine/hooks page do you have in package.json?

6.0.8

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/loading-button-still-interactive-bug-o2ttnb?file=/src/App.tsx

### Do you know how to fix the issue

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

My idea for a fix would be to actually set the disabled attribute on the button when it is loading. Care should be taken not to influence the visual styles though.

**Comment by mustaphaboudouch** [comment]
I also noticed in your code a small issue, when focus on the button and press `SPACE`, the button clicked and animated, but when pressed `ENTER`, it clicked without animation.

**Comment by mustaphaboudouch** [comment]
I think you don't need to conditionally pass onClick.

To prevent the button from receiving focus through keyboard tab navigation, you can simply specify `tabIndex="-1"` on loading.

```html
<button tabIndex={loading ? '-1' : undefined}>...</button>
```

But, I think the best practice is to disable the button on loading, is the same approach used by Material UI and Chakra UI.

- https://mui.com/material-ui/react-button/#loading-button
- https://chakra-ui.com/docs/components/button/usage#button-loading-state

```html
<button disabled={disabled || loading}>...</button>
```

The problem with the second approach is that you can't differentiate between 'loading' styles and 'disabled' styles. However, you can solve this problem by adding a class name to the loading button (for example: '.loading').

```html
<button disabled={disabled || loading} classname={loading ? 'loading' : undefined}>...</button>
```
```css
/* disabled styles */
button.loading:not(:disabled) { ... }
/* loading styles */
button.loading { ... }
```

----------------

In regards the button animation, you are correct.
It's a problem related to the HTML button (the active state is not triggered when we press "ENTER").

**Comment by rtivital** [comment]
This issue is closed for one of these reasons:

- It was fixed in 7.0
- It is no longer reproducible in 7.0
- It is no longer applicable in 7.0
- It does not have a reproduction

If you think that this issue was closed by mistake, and it is still an issue in version 7.0, please reopen it.
